### PR TITLE
update compliance history based on events

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Refer: https://issues.redhat.com/browse/ACM-5453

- Event table
```bash
  » select="select policy_id,cluster_id,created_at,compliance from "event".local_policies ORDER BY policy_id, cluster_id, created_at ASC"
 » kubectl exec -it $(kubectl get pods -n hoh-postgres -l postgres-operator.crunchydata.com/role=master -o jsonpath='{.items..metadata.name}') -c database -n hoh-postgres -- psql -U postgres -d hoh -c "$select"
              policy_id               |              cluster_id              |     created_at      |  compliance
--------------------------------------+--------------------------------------+---------------------+---------------
 74b4b93a-fe9c-4754-bd29-0e986e3b0792 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-25 09:30:13 | non_compliant
 74b4b93a-fe9c-4754-bd29-0e986e3b0792 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-25 09:30:17 | non_compliant
 74b4b93a-fe9c-4754-bd29-0e986e3b0792 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-25 09:31:20 | non_compliant
 74b4b93a-fe9c-4754-bd29-0e986e3b0792 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-25 09:31:29 | compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba01 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 01:52:00 | compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba01 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 03:53:13 | compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba01 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 03:53:16 | non_compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba02 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 01:52:00 | compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba02 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 03:53:13 | non_compliant
 b0c16cfe-ac73-4dde-aa54-5d27feccba02 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24 03:53:16 | compliant
(10 rows)
```
- Compliance History
```bash
» sql="select * from local_status.compliance_history"
» kubectl exec -it $(kubectl get pods -n hoh-postgres -l postgres-operator.crunchydata.com/role=master -o jsonpath='{.items..metadata.name}') -c database -n hoh-postgres -- psql -U postgres -d hoh -c "$sql"
                  id                  |              cluster_id              | compliance_date |  compliance   | compliance_changed_frequency
--------------------------------------+--------------------------------------+-----------------+---------------+------------------------------
 b0c16cfe-ac73-4dde-aa54-5d27feccba01 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24      | non_compliant |                            1
 b0c16cfe-ac73-4dde-aa54-5d27feccba02 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-24      | non_compliant |                            2
 74b4b93a-fe9c-4754-bd29-0e986e3b0792 | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | 2023-05-25      | non_compliant |                            1
(3 rows)
```
- Job log
```bash
2023-05-26 02:38:31 UTC INFO    local-compliance-history        insert success  {"history": "2023-05-25", "inserted": 1, "offset": 0, "batchSize": 1000}
2023-05-26 02:38:31 UTC INFO    local-compliance-history        finish running  {"history": "2023-05-25", "totalCount": 1, "insertedCount": 1, "nextRun": "2023-05-26 02:39:26"}
2023-05-26 02:38:41 UTC INFO    managed-cluster-labels-status-watcher   trimming interval has been reset to 10s
2023-05-26 02:39:11 UTC INFO    managed-cluster-labels-status-watcher   trimming interval has been reset to 20s
2023-05-26 02:39:26 UTC INFO    local-compliance-history        start running   {"history": "2023-05-24", "currentRun": "2023-05-26 02:39:26"}
2023-05-26 02:39:31 UTC INFO    local-compliance-history        insert success  {"history": "2023-05-24", "inserted": 2, "offset": 0, "batchSize": 1000}
2023-05-26 02:39:31 UTC INFO    local-compliance-history        finish running  {"history": "2023-05-24", "totalCount": 2, "insertedCount": 2, "nextRun": "2023-05-26 02:40:26"}
```

### CODE REVIEW

File: test/resources/policy/local-inform-limitrange-policy-placement.yaml [hunk 1]
```diff
@@ -72,18 +72,18 @@ spec:
         matchLabels:
           local-policy-placement: test
 ---
-apiVersion: cluster.open-cluster-management.io/v1beta1
+apiVersion: cluster.open-cluster-management.io/v1beta2
```
Comment (clyang82) [inline test/resources/policy/local-inform-limitrange-policy-placement.yaml:75]: Do we add the managedcluster into default managedclusterset? Here suggest to use `global` managedclusterset. it is include all managedcluster by default.

File: manager/pkg/cronjob/task/compliance_history.go [hunk 2]
```diff
@@ -36,51 +40,55 @@ func SyncLocalCompliance(ctx context.Context, pool *pgxpool.Pool, enableSimulati
...
 	// if err != nil {
 	// 	return totalCount, syncedCount, err
 	// }
-
-	err = pool.QueryRow(ctx, fmt.Sprintf("SELECT COUNT(*) FROM %s", viewName)).Scan(&totalCount)
-	if err != nil {
+	totalCountSQLTemplate := `
+		SELECT COUNT(*) FROM (
+			SELECT DISTINCT policy_id, cluster_id FROM event.local_policies
+			WHERE created_at BETWEEN CURRENT_DATE - INTERVAL '%d days' AND CURRENT_DATE - INTERVAL '%d day'
```
Comment (clyang82) [inline manager/pkg/cronjob/task/compliance_history.go:70]: Do you mean that `BETWEEN 2023-05-25 AND 2023-05026` get the data for 2023-05-25?

**Review by clyang82 — COMMENTED** [review]
/assign @morvencao to look at the simulation part changes.

File: manager/pkg/cronjob/task/compliance_history.go (hunk 2 above)
Comment (yanmxa) [reply manager/pkg/cronjob/task/compliance_history.go:70]: Yes~

File: test/resources/policy/local-inform-limitrange-policy-placement.yaml (hunk 1 above)
Comment (yanmxa) [reply test/resources/policy/local-inform-limitrange-policy-placement.yaml:75]: Sure!
Done~

**Review by morvencao — COMMENTED** [review]
💯 
/lgtm

**Pull request merged**